libraries/chain/include/eosio/chain/database_utils.hpp

```diff
@@ -4,96 +4,73 @@
  */
 #pragma once
 
-#include <eosio/chain/types.hpp>
-#include <fc/io/raw.hpp>
 #include <softfloat.hpp>
 
+#include <fc/io/raw.hpp>
+
+#include <cyberway/chaindb/controller.hpp>
+
 namespace cyberway { namespace chaindb {
     class chaindb_controller;
 } } // namespace cyberway::chaindb
 
 namespace eosio { namespace chain {
 
-   template<typename ...Indices>
-   class index_set;
+   template<typename ...Tables>
+   class table_set;
 
-   template<typename Index>
-   class index_utils {
+   template<typename Table>
+   class table_utils {
+      using chaindb_controller = cyberway::chaindb::chaindb_controller;
       public:
-         using index_t = Index;
-
          template<typename F>
-         static void walk( const chainbase::database& db, F function ) {
-            auto const& index = db.get_index<Index>().indices();
-            const auto& first = index.begin();
-            const auto& last = index.end();
+         static void walk(const chaindb_controller& chaindb, const F& function) {
+            const auto table = Table(chaindb);
+            const auto& first = table.begin();
+            const auto& last = table.end();
             for (auto itr = first; itr != last; ++itr) {
                function(*itr);
             }
          }
 
-         template<typename Secondary, typename Key, typename F>
-         static void walk_range( const chainbase::database& db, const Key& begin_key, const Key& end_key, F function ) {
-            const auto& idx = db.get_index<Index, Secondary>();
-            auto begin_itr = idx.lower_bound(begin_key);
-            auto end_itr = idx.lower_bound(end_key);
-            for (auto itr = begin_itr; itr != end_itr; ++itr) {
-               function(*itr);
-            }
-         }
-
-         template<typename Secondary, typename Key>
-         static size_t size_range( const chainbase::database& db, const Key& begin_key, const Key& end_key ) {
-            const auto& idx = db.get_index<Index, Secondary>();
-            auto begin_itr = idx.lower_bound(begin_key);
-            auto end_itr = idx.lower_bound(end_key);
+         template<typename Key>
+         static size_t size_range(const chaindb_controller& chaindb, const Key& begin_key, const Key& end_key ) {
```

Comment:
This function is really need?
For what?

Reply:
walk range method has been removed. This method was below. I have fixed it and don't check if it really needed

Reply:
This method is not used. Wil be removed